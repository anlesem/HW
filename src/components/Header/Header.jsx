import style from './Header.module.scss';

import { HeaderNavBar } from '../HeaderNavBar/HeaderNavBar';

export const Header = () => {
  return (
    <div className={style.wrap + ' container'}>
      <HeaderNavBar />
    </div>
  );
};
