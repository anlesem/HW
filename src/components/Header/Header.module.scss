.header {
	width: 100%;
	height: 100px;
	background-color: #999;

	&__wrap {
		display: grid;
		place-items: center;
		height: 100%;
	}

	h1 {
		font-size: 24px;
		font-weight: bold;
	}
}