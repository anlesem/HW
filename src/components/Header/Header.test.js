import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { Header } from './Header';

describe('Header', () => {
  it('Компонент существует', () => {
    expect(Header).toBeInstanceOf(Function);
  });

  it('Снимок состояния', () => {
    const { asFragment } = render(
      <BrowserRouter>
        <Header />
      </BrowserRouter>
    );
    expect(asFragment(<Header />)).toMatchSnapshot();
  });
});
