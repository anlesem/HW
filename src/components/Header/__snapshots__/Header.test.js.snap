// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Снимок состояния 1`] = `
<DocumentFragment>
  <div
    class="wrap container"
  >
    <ul
      class="navbar"
    >
      <li>
        <a
          aria-current="page"
          class="active"
          data-testid="NavLink-1"
          href="/"
        >
          <button>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium css-1q8v7ak-MuiSvgIcon-root"
              data-testid="AccountCircleTwoToneIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 4c-4.41 0-8 3.59-8 8 0 1.82.62 3.49 1.64 4.83 1.43-1.74 4.9-2.33 6.36-2.33s4.93.59 6.36 2.33C19.38 15.49 20 13.82 20 12c0-4.41-3.59-8-8-8zm0 9c-1.94 0-3.5-1.56-3.5-3.5S10.06 6 12 6s3.5 1.56 3.5 3.5S13.94 13 12 13z"
                opacity=".3"
              />
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zM7.07 18.28c.43-.9 3.05-1.78 4.93-1.78s4.51.88 4.93 1.78C15.57 19.36 13.86 20 12 20s-3.57-.64-4.93-1.72zm11.29-1.45c-1.43-1.74-4.9-2.33-6.36-2.33s-4.93.59-6.36 2.33C4.62 15.49 4 13.82 4 12c0-4.41 3.59-8 8-8s8 3.59 8 8c0 1.82-.62 3.49-1.64 4.83zM12 6c-1.94 0-3.5 1.56-3.5 3.5S10.06 13 12 13s3.5-1.56 3.5-3.5S13.94 6 12 6zm0 5c-.83 0-1.5-.67-1.5-1.5S11.17 8 12 8s1.5.67 1.5 1.5S12.83 11 12 11z"
              />
            </svg>
            Профиль
          </button>
        </a>
      </li>
      <li>
        <a
          class=""
          data-testid="NavLink-2"
          href="/chats"
        >
          <button>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium css-1q8v7ak-MuiSvgIcon-root"
              data-testid="ChatBubbleTwoToneIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="m4 18 2-2h14V4H4z"
                opacity=".3"
              />
              <path
                d="M20 2H4c-1.1 0-2 .9-2 2v18l4-4h14c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zm0 14H6l-2 2V4h16v12z"
              />
            </svg>
            Чаты
          </button>
        </a>
      </li>
      <li>
        <a
          class=""
          data-testid="NavLink-3"
          href="/data"
        >
          <button>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium css-1q8v7ak-MuiSvgIcon-root"
              data-testid="CloudQueueTwoToneIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 12h-1.5v-.5C17.5 8.46 15.04 6 12 6c-2.52 0-4.63 1.69-5.29 4H6c-2.21 0-4 1.79-4 4s1.79 4 4 4h13c1.66 0 3-1.34 3-3s-1.34-3-3-3z"
                opacity=".3"
              />
              <path
                d="M19.35 10.04C18.67 6.59 15.64 4 12 4 9.11 4 6.6 5.64 5.35 8.04 2.34 8.36 0 10.91 0 14c0 3.31 2.69 6 6 6h13c2.76 0 5-2.24 5-5 0-2.64-2.05-4.78-4.65-4.96zM19 18H6c-2.21 0-4-1.79-4-4s1.79-4 4-4h.71C7.37 7.69 9.48 6 12 6c3.04 0 5.5 2.46 5.5 5.5v.5H19c1.66 0 3 1.34 3 3s-1.34 3-3 3z"
              />
            </svg>
            Данные
          </button>
        </a>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
